jobserver/views/job_requests.py

```diff
@@ -109,16 +110,22 @@ def get_queryset(self):
                 # it then we can look for a job number
                 qs = qs.filter(qwargs | Q(jobs__pk=q))
 
+        def is_an_int(value):
+            try:
+                return int(value)
+            except ValueError:
+                raise BadRequest
+
         backend = self.request.GET.get("backend")
-        if backend:
+        if backend and is_an_int(backend):
```

Comment:
Nice, I think this is much better. Only minor nit is that the conditional is now slightly misleading because it looks like it will just skip the block if `backend` isn't an int, rather than raising an error.

Even more minor nit is that `is_an_int("0")` will evaluate as false rather than true (because it returns 0) :smile:

Could you instead have?
```py
if backend:
    raise_if_not_int(backend)
    qs = qs.filter(backend_id=backend)
```

Reply:
Arg yes, clearly didn't pay attention to the if 🤦‍♂️

Updated as suggested 👍